src/OmniSharp.Roslyn/BufferManager.cs

```diff
@@ -46,8 +56,12 @@ public async Task UpdateBufferAsync(Request request)
 
             var solution = _workspace.CurrentSolution;
 
-            var documentIds = solution.GetDocumentIdsWithFilePath(request.FileName);
-            if (!documentIds.IsEmpty)
+            // To properly update the buffer wait until all projects are loaded.
+            var documentIds = inFullProjectModel ? 
+                await _workspace.GetDocumentIdsWithFilePathFromFullProjectModelAsync(request.FileName) :
```

Comment:
This is...concerning. What happens if the queue has a lot of stuff in it and buffer update requests start piling up?

Reply:
@rchande : the updates will wait. Without this I've seen lot of errors generated reported for a file while projects are still being loaded. Let me try to get a stable repro for this through and get back to you.

Reply:
Reverted this part of the change because it turns out even without any of my changes the error repros. Opened [https://github.com/OmniSharp/omnisharp-roslyn/issues/1356](https://github.com/OmniSharp/omnisharp-roslyn/issues/1356)

---
In reply to: [239610163](https://github.com/OmniSharp/omnisharp-roslyn/pull/1322#discussion_r239610163) [](ancestors = 239610163)

Reply:
... and confirmed that the fix for the bug works when projects are loaded on demand as well.

---
In reply to: [240308562](https://github.com/OmniSharp/omnisharp-roslyn/pull/1322#discussion_r240308562) [](ancestors = 240308562,239610163)